lib/TImerManager.cs

```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+
+namespace SpeedMod.Timer {
+    public class TimerManager : MonoBehaviour {
+        private LevelStats levelStatsInstance;
+        public float GetStartTime() {    
+            LevelStats[] levelStatsInstances = FindObjectsOfType<LevelStats>();
+            levelStatsInstance = levelStatsInstances[0];
+            return levelStatsInstance.startTime;
+        }
+
+        public void UpdateTimer(float time, float startTime) {
+            LevelStats[] levelStatsInstances = FindObjectsOfType<LevelStats>();
+            levelStatsInstance = levelStatsInstances[0];
+            levelStatsInstance.startTime = Time.time - (time - startTime);
+        }
```

Comment:
This can be updated with the changes recommended in the `Checkpoints.cs` comments:
```suggestion
        public void UpdateTimer(float checkpointTime) {
            LevelStats[] levelStatsInstances = FindObjectsOfType<LevelStats>();
            levelStatsInstance = levelStatsInstances[0];
            levelStatsInstance.startTime = Time.time - checkpointTime;
        }
```